lang/py/avro/test/test_compatibility.py

```diff
@@ -659,6 +688,7 @@ def test_schema_compatibility(self):
                 ENUM_ABC_FIELD_DEFAULT_B_ENUM_DEFAULT_A_RECORD,
             ),
             (NS_RECORD1, NS_RECORD2),
+            (WITHOUT_NAMESPACE_RECORD, WITH_NAMESPACE_RECORD),
```

Comment:
Feel free to ignore this as I'm not too familiar with this code, but are these two really compatible? To me, it reads like one of them has the null namespace and the other one has the namespace "ns". Don't namepaces have to match to be compatible?

I'd expect `{"namespace":"ns","name":"Record"}` to be compatible with `{"name":"ns.Record"}` but not too sure about if that's compatible with `{"name":"Record"}`. But like I said, I'm not an expert on Avro.

Reply:
From the specification, https://avro.apache.org/docs/1.11.1/specification/_print/#schema-resolution
```
* It is an error if the two schemas do not match. To match, one of the following must hold:
  ...
  * both schemas are records with the same (unqualified) name
  ...
```
I am assuming this holds for compatibility.

This change also aligns Python lib behavior with the Java implementation.
The compatibility of names is handled on this line in Python lib: https://github.com/apache/avro/blob/master/lang/py/avro/compatibility.py#L378
In Java lib it is here: https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/SchemaCompatibility.java#L97

It could be error in the specification and therefore in the implementation that namespace is not relevant.

Reply:
Thanks for explaining, it does look like this is per the spec then.